buf/registry/module/v1beta1/tag_service.proto

```diff
@@ -95,18 +101,26 @@ message ListTagsResponse {
 message CreateTagsRequest {
   // An individual request to create a Tag.
   message Value {
-    // The Module to create the Tag under.
-    ModuleRef module_ref = 1 [(buf.validate.field).required = true];
     // The Tag name.
-    string name = 2 [
+    //
+    // If there is already a Tag with a name, the request errors. Otherwise, a new Tag will be
+    // created associated with the commit identified.
+    repeated string name = 1 [
       (buf.validate.field).required = true,
       (buf.validate.field).string.max_len = 250
     ];
-    // The id of the Commit associated with the Tag.
-    string commit_id = 3 [
-      (buf.validate.field).required = true,
-      (buf.validate.field).string.uuid = true
-    ];
+    // The reference to resolve the Commit to which these Tags will be set.
+    //
+    // See the documentation on Ref for resource resolution details.
+    //
+    // The Tags are set depending on the kind of resource that is resolved:
+    //   - If a Module is referenced, Tags are set to the latest released Commit in the module.
+    //   - If a Commit is referenced, Tags are set for that Commit.
+    //   - If a Tag is referenced, Tags are set for the Commit that this Tag references.
+    //   - If a VCSCommit is referenced, Tags are set for the Commit that this VCSCommit references.
+    //   - Is a Branch is referenced, Tags are set for the latest Commit on the Branch.
+    //   - If a Digest is referenced, Tags are set for the most recent Commit with this Digest.
+    ResourceRef resource_ref = 2 [(buf.validate.field).required = true];
```

Comment:
Why do we need to do this? Why can't I get the commit first?

Reply:
You could resolve the commit id from the VCS ref, yes, but that's one more query the client could save if they pass the `ResourceRef` directly here... although as @saquibmian mentions, that takes away the idempotent property because the ref could change across requests.

Reply:
Chatted with Peter, I'll update this.